### PR TITLE
[CENNSO-799] X509 certificate validity support

### PR DESCRIPTION
**Pull request by td2501**
Add support for X509 certificates validity.

[jira](https://travelping.atlassian.net/browse/CENNSO-799)

### CODE REVIEW

File: strongswan/collector.go [hunk 1]
```diff
@@ -475,3 +512,67 @@ func viciStateToInt(v string) connectionStatus {
...
+		cert, err := x509.ParseCertificate([]byte(data))
+		if err != nil {
+			return nil, err
+		}
+
+		res = append(res, Crt{
+			Subject:   cert.Subject.String(),
+			NotBefore: cert.NotBefore,
+			NotAfter:  cert.NotAfter,
+		})
```
Comment (torilabs) [inline strongswan/collector.go:574]: I would rather parse all the attributes in a raw form. Then you can filter and process `X509` type later in  `collectX509CrtMetrics`.

File: strongswan/collector.go [hunk 2]
```diff
@@ -475,3 +512,67 @@ func viciStateToInt(v string) connectionStatus {
 		return unknown
 	}
 }
+
+func (c *Collector) collectCrtMetrics(crt *Crt, ch chan<- prometheus.Metric) {
```
Comment (torilabs) [inline strongswan/collector.go:516]: unit tests are missing

**Review by torilabs — COMMENTED** [review]
Please don't use your internal commit prefix `[CENNSO-799]` in this project.

**Comment by td2501** [comment]
This was draft created on wrong fork. I will close it for now as I'm still working on it.

Thanks for all the comments. I will apply them in final version and create new PR.

**Pull request closed without merging**